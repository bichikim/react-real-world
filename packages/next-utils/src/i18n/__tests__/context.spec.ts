import {render} from '@testing-library/react'
import {createElement as h, useContext} from 'react'
import {createRosetta, I18nContext} from '..'

describe('i18n context', function test() {
  describe('createRosetta', function test() {
    it('should create namespace rosetta ', function test() {
      const {set, locale, __t, t} = createRosetta()
      locale('ko-KR')
      set('ko-KR', {
        bar: ['바'],
        f: {
          o: {
            o: '포오',
          },
        },
        foo: '포',
      })
      expect(t('foo')).toBe('포')
      expect(t.add('f.o')('o')).toBe('포오')
      expect(t.raw('bar')).toEqual(['바'])
      expect(__t('foo')).toBe('포')
    })
  })

  describe('I18nContext', function test() {
    it('should be an i18n context', function test() {
      const context = I18nContext

      const Component = () => {
        const i18n = useContext(context)
        return (
          h('div', null, i18n.t('foo'))
        )
      }

      const {container} = render(h(Component))

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
