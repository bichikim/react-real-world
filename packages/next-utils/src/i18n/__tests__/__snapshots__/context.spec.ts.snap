// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n context I18nContext should be an i18n context 1`] = `
<div>
  foo
</div>
`;
